### ISSUE TITLE
bug: Input popups (autofill, scan new credit card, etc) not opening on Android

### ISSUE THREAD

**Issue by coltontravers** [ionic-team/ionic-framework#23695]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

The android app is not displaying any of the autofill (name, email, etc) and other input popups on Android. I have tried on both an emulator and on a physical device. It does however work correctly if I open it in chrome (on both the emulator and physical device).

The inputs also do work correctly on IOS.

### Expected Behavior

Android autofill popups should open when clicking on an input with correct attributes.

### Steps to Reproduce

Add an input like below. I have tested with various attributes, etc to no avail.

- `<ion-input name="lname" autocomplete="family-name" type="text" formControlName="lastName" autocapitalize="words"></ion-input>`

**Comment by liamdebeasi** [comment]
Thanks for the issue. What version of the Android webview are you using in that application? You can check by logging `window.navigator.userAgent` in the console when inspecting the app.

**Comment by coltontravers** [comment]
@liamdebeasi I'm getting "Mozilla/5.0 (Linux; Android 11; sdk_gphone_x86_arm Build/RSR1.201013.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/83.0.4103.106 Mobile Safari/537.36" for that.

**Comment by liamdebeasi** [comment]
Thanks. Can you reproduce this issue in an Ionic starter app and provide a link to the repo? I cannot reproduce this on my end.

**Comment by coltontravers** [comment]
@liamdebeasi Sure, replaced the login form in the kitchen sink app with a "family-name" input that doesn't popup the autofill on android here: https://github.com/coltontravers/autofill-bug

Also, for more context here's a screenshot of it not display in the emulator, and with it displaying in chrome
![image](https://user-images.githubusercontent.com/6713494/127245437-90324b9e-47a5-421d-8d65-b2683da5b5fa.png)
![image](https://user-images.githubusercontent.com/6713494/127245580-d4d5ab72-37eb-43ad-8805-c63ab6196e8c.png)

**Comment by coltontravers** [comment]
@liamdebeasi Was wondering if you had a moment to look at this? The other person on my project is also seeing this issue. Thanks!

**Comment by liamdebeasi** [comment]
Apologies for the delay. I have been out of the office for several days and am getting caught up. I will try to take a look later this week.

**Comment by coltontravers** [comment]
Alright, thanks!

**Comment by coltontravers** [comment]
Oh, I also wanted to point out the autocomplete doesn't seem to work on IOS for a "given-name" autocomplete input. All other versions of name seem to work fine though (name, family-name, etc).

**Comment by liamdebeasi** [comment]
Do these issues happen if you use a native `<input>` instead of `<ion-input>`?

**Comment by coltontravers** [comment]
Yes, they do.

**Comment by liamdebeasi** [comment]
If issue persists with just `<input>`, then this is likely a bug in Chromium not WebKit. Are you able to create a reproduction of this issue without using Ionic Framework? We can look into reporting this to the appropriate teams so the issues can get fixed.

**Comment by coltontravers** [comment]
Sure, what would you like the reproduction with other than Ionic? It does work fine when loading the Ionic app in Chrome like I said earlier in the issue, btw.

**Comment by liamdebeasi** [comment]
Given that this is happening only in a webview, I will usually create a blank Capacitor app and add my code in the app.

Here is the starter app I use: https://github.com/liamdebeasi/blank-webview-repro.

You can add your code to `www/index.html` and then run the usual Capacitor commands to test it:

1. Install dependencies: `npm install`.
2. Add the android platform: `npx cap add android`.
3. Open with Android Studio: `npx cap open android`.

I normally would do this, but since I am having a hard time reproducing the issue on my end if you could add your minimal code reproduction and then push to a GitHub repo, I can test it again. From there we can move forward with reporting this to the Chromium team.

**Comment by coltontravers** [comment]
Alright, I pushed up a repo using that and with a very basic form. The interesting thing is that the inputs do now autofill correctly for Android there, so it's confusing as to why the Ionic app above isn't working (at least for my team and I). Also, the IOS issue of the "given-name" input not working with autofill is still happening in this repo. Repo: https://github.com/coltontravers/autofill-blank

**Comment by willmartindev** [comment]
Hi @coltontravers,

We think this is a bug with Android WebView. I created a ticket with the Chromium team [here.](https://bugs.chromium.org/p/chromium/issues/detail?id=1238338)

Once we receive feedback on that, I will make sure to update this ticket as well so that you are notified. Thanks again!

**Comment by coltontravers** [comment]
Alright, thank you!

**Comment by coltontravers** [comment]
@willmartindev @liamdebeasi I think I actually found the issue for no autofill showing on Android. I'm now getting this error: "W/chromium: [WARNING:autofill_agent.cc(789)] Autofill suggestions are disabled because the document isn't a secure context.". The odd thing is it doesn't popup that warning every time in android studio - sometimes it requires me to restart the app multiple times for that warning to show. I tried running ionic with the --ssl flag, but that whitescreens and complains about certs it looks like even though I added/installed them.

**Comment by liamdebeasi** [comment]
Hey there,

Thanks for your patience while we investigated this. It turns out that the Chromium team removed support for autocomplete in the Android WebView (see: https://bugs.chromium.org/p/chromium/issues/detail?id=1238338#c15) in favor of developers integrating with the Android autocomplete framework in their apps (see: https://bugs.chromium.org/p/chromium/issues/detail?id=1238338#c17).

At this point there is not much that Ionic Framework can control, so I think it makes sense for the Capacitor team to investigate adding support for the Android autocomplete framework so that any Capacitor app can have autocomplete capabilities.

I am going to close this issue since there is nothing actionable from the Ionic Framework side, but I created a feature request with the Capacitor team for this addition (https://github.com/ionic-team/capacitor/issues/5052). I recommend following https://github.com/ionic-team/capacitor/issues/5052 for any updates on this issue.

Thank you!
